ui/server/src/main/scala/pl/touk/nussknacker/ui/api/UserResources.scala

```diff
@@ -13,14 +13,29 @@ class UserResources(typesForCategories: ProcessTypesForCategories)(implicit ec:
     path("user") {
       get {
         complete {
-          DisplayableUser(
-            id = user.id,
-            isAdmin = user.isAdmin,
-            categories = typesForCategories.getAllCategories,
-            categoryPermissions = user.categoryPermissions.mapValues(_.map(_.toString)))
+          DisplayableUser(user, typesForCategories.getAllCategories)
         }
       }
     }
 }
 
-@JsonCodec case class DisplayableUser(id: String, isAdmin: Boolean, categories: List[String], categoryPermissions: Map[String, Set[String]])
\ No newline at end of file
+@JsonCodec case class DisplayableUser private(id: String, isAdmin: Boolean, categories: List[String], categoryPermissions: Map[String, Set[String]])
+
+object DisplayableUser {
+  import pl.touk.nussknacker.engine.util.Implicits._
+
+  def apply(user: LoggedUser, allCategories: List[String]): DisplayableUser = user match {
+    case CommonUser(id, categoryPermissions) => new DisplayableUser(
+      id = id,
+      isAdmin = false,
```

Comment:
false is default, you don't need set it.

Reply:
it's not and btw, I'd prefer to make it explicit